integration-testing/src/tools/fixture.py

```diff
@@ -1,22 +1,12 @@
 import pytest
 import inspect
 
-# def wrapper(request, a,b,c):
-#     import inspect
-#     fixture_values = dict((f,request.getfuncargvalue(f.__name__)) for f in [a,b])
-#     def get_value(p):
-#         if p in fixture_values:
-#             return request.getfuncargvalue(f.__name__)
-#         else:
-#             return p
-#     parameter_values = [get_value(param_name) for param_name in [a,b,c]]
-#     return fn(*param_values)
 
 def make_wrapper(fn, fixtures):
     parameter_names = inspect.signature(fn).parameters.keys()
     parameter_list = ",".join(p for p in parameter_names if p != 'request')
-    parameter_name_list = ",".join(f"('{p}', {p})" for p  in parameter_names)
-    namespace = {"fn":fn, "fixtures":fixtures}
+    parameter_name_list = ",".join(f"('{p}', {p})" for p in parameter_names)
+    namespace = {"fn": fn, "fixtures": fixtures}
 
     wrapper_code = f"""
 def {fn.__name__}(request, {parameter_list}):
```

Comment:
I'm not sure what you mean. Can you be more specific?

Reply:
So you mean that I should create a piece of Python AST and then execute it? I tried that but it was too complicated and less readable than the current approach

Reply:
Also this is not in the scope of the current pr. So we should move forward with this PR and maybe address the problem later

Reply:
Also this is not in the scope of the current pr. So we should move forward with this PR and maybe address the problem later

Reply:
> So you mean that I should create a piece of Python AST and then execute it? I tried that but it was too complicated and less readable than the current approach

Something simpler:  Instead of `exec('def foo...')` you can do something like

```
class fn():
  def __init__(self, **kwargs):
    self.kwargs = kwargs

  def __call__(self, args):
    BODY IS self.kwargs

return fn(args know at definition time)
```

> Also this is not in the scope of the current pr. So we should move forward with this PR and maybe address the problem later

Fair enough.  Looking forward to the PR.